development-docs/TESTING.md

```diff
@@ -32,50 +32,51 @@ The next requirement is to have built the `systemtest` package dependencies, whi
+The current lifecycle mechanism will ensure that all resources created within these classes will be deleted after finished tests.
 
-And classes: 
+Notable classes:
 
-* **Environment** — singleton class, which loads the test environment variables (see following section), which are used in the tests.
-* **Constants** — simple interface holding all constants used in the tests. 
-* **resources/ResourceManager** - singleton class which store info about deployed resources and take care about proper resource deletion.  
+* **Environment** — singleton class, which loads the test environment variables (see following section), which are used in tests.
+* **Constants** — simple interface holding all constants used in tests.
+* **resources/ResourceManager** - singleton class which stores data about deployed resources and takes care of proper resource deletion.
 
 ## Test Phases
 
-In general, we have classic test phases: setup, exercise, test, teardown.
+In general, we use classic test phases: `setup`, `exercise`, `test` and `teardown`.
 
-#### Setup
+### Setup
 
-In this phase we do the following things:
+In this phase we perform:
 
 * Create namespace(s)
 * Deploy the Strimzi Cluster operator
-* Deploy a Kafka cluster or other components (optional)
+* Deploy the Kafka cluster and/or other components (optional)
 
-The reason why the last point is optional is because we have some test cases where you want to have a different kafka configuration for each test scenario so creation of the Kafka cluster and other resources is done in the test phase.
+The reason why the last point is optional, is because we have some test cases where you want to have a different kafka configuration for each test scenario, so creation of the Kafka cluster and other resources is done in the test phase.
 
-We create resources in Kubernetes cluster via classes in `resources` package, which allows you to deploy all components and, if needed, change them from their default configuration using a builder. 
-Currently, we have two stacks, which are stored in `ResourceManager` singleton instance — one for whole test class resources and one for test method resources.
-You can create resources anywhere you want, and our implementation will take care about put resource on top of one stack and delete them at the end of test method/class.
+We create resources in Kubernetes cluster via classes in `resources` package, which allows you to deploy all components and, if needed, change them from their default configuration using a builder.
+Currently, we have two stacks, which are stored in `ResourceManager` singleton instance — one for all test class resources and one for test method resources.
+You can create resources anywhere you want. Our resource lifecycle implementation will handle insertion of the resource on top of stack and deletion at the end of the test method/class.
```

Comment:
Just thinking - do we really want almost same sentence so many times? It's the third time until this line when I see this - `... resource lifecycle implementation will ... and deletion of resources at the end of tests`. 

Do we want this so many times in our docu? It's just idea.

Reply:
I think I can simplify it on lines  49-50. Let me have a look at it.